### DISCUSSION TITLE
PHP 8 only support

### DISCUSSION THREAD

**Discussion by antonmedv** [deployphp/deployer#3429] [General]
I'm thinking to ditch php7 support and release a new major Deployer v8 version with the requirement on >= php8.0.

WDYT?

**Comment by benoitchantre** [comment]
Why not to wait a bit and see if other breaking changes would be needed. Deployer v7 release is only 6 months old.
Does it bring a huge benefit for the maintenance?

**Reply by antonmedv** [reply]
Good point. We can t other breaking changes.

**Reply by antonmedv** [reply]
On another hand all our deps already only php8. So it’s difficult to upgrade.

**Reply by codeit-ninja** [reply]
It does encourage people to use a supported version of php instead if eol versions :)

**Comment by GlennM** [comment]
Just my 2 cents:
PHP 7.4 is EOL (including security fixes), so people should be encouraged to upgrade to a more secure version. In the end we all benefit from using more secure applications.

**Comment by codeit-ninja** [comment]
The value inside the composer.json for PHP is a bit odd right now, `^7.3|^8.0`. Isnt the correct syntax for this `^7.3 || ^8.0`?

I would indeed just ditch php 7 and go for php 8 because php 7 is EOL.

**Comment by benoitchantre** [comment]
I think that would be a good idea increase the minimal supported version to ease the maintenance.
